### DISCUSSION TITLE
What is the difference between `TensorOp` and `WmmaTensorOp`?

### DISCUSSION THREAD

**Discussion by alexarmbr** [NVIDIA/cutlass#1446] [Q&A] [answered]
In [this](https://github.com/NVIDIA/cutlass/blob/main/media/docs/functionality.md) page, in the column for "Opcode Class" in the listing of CUTLASS kernels, two of the options are `TensorOp` and `WmmaTensorOp`, I am wondering if someone could distinguish these for me.

I assume `WmmaTensorOp` uses (nvcuda::wmma)[https://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html#warp-matrix-functions] to access tensor cores? If this is true then how to the `TensorOp` kernels access Tensor Cores?

**Comment by hwu36** [comment, marked as answer]
`WmmaTensorOp` uses `wmma.mma`, and `TensorOp` uses `mma` in this [ptx doc](https://github.com/NVIDIA/cutlass/blob/main/media/docs/functionality.md) .  The former is easier to use, the latter is faster.
